benchflow-dsl/src/main/scala/cloud/benchflow/dsl/definition/yamlprotocol/valuesrange/IntValuesRangeYamlProtocol.scala

```diff
@@ -0,0 +1,103 @@
+package cloud.benchflow.dsl.definition.yamlprotocol.valuesrange
+
+import cloud.benchflow.dsl.definition.configuration.goal.explorationspace.step.RangeWithStep
+import cloud.benchflow.dsl.definition.errorhandling.BenchFlowDeserializationExceptionMessage
+
+import scala.util.{Failure, Success, Try}
+
+/**
+  * @author Vincenzo Ferme <info@vincenzoferme.it>
+  */
+trait IntValuesRangeYamlProtocol extends ValuesRangeYamlProtocol {
+
+  protected override def getStepList(range: List[Int], step: String): Try[List[Int]] = {
+
+    val rangeValid = assertRangeIsValid(range)
+
+    if (rangeValid.isFailure) {
+      Failure(rangeValid.failed.get)
+    } else {
+
+      val (prefix: String, stepValue: Int) = getIntStep(step)
+
+      val tryStepList = generateStepList(range, stepValue, prefix)
+
+      if(tryStepList.isSuccess)
+        Success(tryStepList.get)
+      else
+        Failure(tryStepList.failed.get)
+
+
+    }
+
+  }
+
+  protected def getIntStep(step: String) = {
+
+    val (prefix: String, triedStepValue: Try[Int]) = tryGetStep(step)
+
```

Comment:
It seems this method is only called once so we could integrate it into this method, also because we explicitly get an int step in the `tryGetStep` method